mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -126,9 +129,10 @@ abstract class Predictor[
    * and copying parameters into the model.
    *
    * @param dataset  Training dataset
+   * @param handlePersistence  Whether to persist the dataset in implementation
    * @return  Fitted model
    */
-  protected def train(dataset: Dataset[_]): M
+  protected def train(dataset: Dataset[_], handlePersistence: Boolean): M
```

Comment:
It's better if we can find a way not to break the API. We can give it some time to see if there's better solution. Also if we have to, can we have some default value for the new parameter?